### ISSUE TITLE
[BUG] 1.91.0 Docker - "operator class 'cosine_ops' does not exist for access method 'vectors'"

### ISSUE THREAD

**Issue by Teonyr** [immich-app/immich#5731]
### The bug

I've updated my Docker compose as it is stated in the current release notes and deployed the required postgres image.
However, immich-server isn't starting properly, its displaying the following error message:

```
[Nest] 7  - 12/15/2023, 7:01:28 PM   ERROR [ExceptionHandler] operator class "cosine_ops" does not exist for access method "vectors"

QueryFailedError: operator class "cosine_ops" does not exist for access method "vectors"

    at PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:211:19)

    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

    at async AddCLIPEmbeddingIndex1700713994428.up (/usr/src/app/dist/infra/migrations/1700713994428-AddCLIPEmbeddingIndex.js:9:9)

    at async MigrationExecutor.executePendingMigrations (/usr/src/app/node_modules/typeorm/migration/MigrationExecutor.js:225:17)

    at async DataSource.runMigrations (/usr/src/app/node_modules/typeorm/data-source/DataSource.js:260:35)

    at async DataSource.initialize (/usr/src/app/node_modules/typeorm/data-source/DataSource.js:148:17)
```

### The OS that Immich Server is running on

Docker

### Version of Immich Server

1.91.0

### Version of Immich Mobile App

1.90.2

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
version: "3.8"

networks:
 default:
  external:
   name: backend

services:
  immich-server:
    container_name: immich-server
    image: altran1502/immich-server:release
    entrypoint: ["./start.sh", "immich"]
    volumes:
      - /media/raid/storage/immich:/usr/src/app/upload
    env_file:
      - stack.env
    ports:
      - 2283:3001
    labels:
      traefik.enable: true
       [REMOVED]
    restart: unless-stopped
    
  immich-microservices:
    container_name: immich-microservices
    image: altran1502/immich-server:release
    entrypoint: ["./start.sh", "microservices"]
    volumes:
      - /media/raid/storage/immich:/usr/src/app/upload
    env_file:
      - stack.env
    restart: unless-stopped
    
  immich-machine-learning:
    container_name: immich-machine-learning
    image: altran1502/immich-machine-learning:release
    #entrypoint: ["/bin/sh", "./entrypoint.sh"]
    volumes:
      - /media/raid/storage/immich:/usr/src/app/upload
    env_file:
      - stack.env
    restart: unless-stopped
```

### Your .env content

```Shell
DB_HOSTNAME=postgres-15
DB_USERNAME=immich
DB_PASSWORD=[REMOVED]
DB_DATABASE_NAME=immich
REDIS_HOSTNAME=redis
JWT_SECRET=[REMOVED]
IMMICH_WEB_URL=http://immich-web:3000
IMMICH_SERVER_URL=http://immich-server:3001
IMMICH_MACHINE_LEARNING_URL=http://immich-machine-learning:3003
NODE_ENV=production
REDIS_PASSWORD=[REMOVED]
TYPESENSE_API_KEY=[REMOVED]
TYPESENSE_DATA_DIR=/data
IMMICH_API_URL_EXTERNAL=[REMOVED]
```

### Reproduction steps

```bash
1. Start server
2. Observe error
...
```

### Additional information

Just for info, this is my postgres database, I'm using for multiple services:
```
  postgres-15:
    #image: postgres:15
    image: tensorchord/pgvecto-rs:pg15-latest
    container_name: postgres-15
    restart: always
    volumes:
      - /persistent/postgres-15:/var/lib/postgresql/data
    environment:
      POSTGRES_PASSWORD: [REMOVED]
```

**Comment by Teonyr** [comment]
Addition:
Changing `image: tensorchord/pgvecto-rs:pg15-latest` to `image: tensorchord/pgvecto-rs:pg15-v0.1.11`, changes the error thrown to:
```
[Nest] 1  - 12/15/2023, 7:35:24 PM   ERROR [ExceptionHandler] could not find function "typmod_in_wrapper" in file "/usr/lib/postgresql/15/lib/vectors.so"

QueryFailedError: could not find function "typmod_in_wrapper" in file "/usr/lib/postgresql/15/lib/vectors.so"

    at PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:211:19)

    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

    at async UsePgVectors1700713871511.up (/usr/src/app/dist/infra/migrations/1700713871511-UsePgVectors.js:20:9)

    at async MigrationExecutor.executePendingMigrations (/usr/src/app/node_modules/typeorm/migration/MigrationExecutor.js:225:17)

    at async DataSource.runMigrations (/usr/src/app/node_modules/typeorm/data-source/DataSource.js:260:35)

    at async DataSource.initialize (/usr/src/app/node_modules/typeorm/data-source/DataSource.js:148:17)
```

**Comment by mertalev** [comment]
I think the first error is because you used the latest tag (don't use this, this is a nightly release). As for the second, it might be because you switched it back to v0.1.11 and there's now a reference to a function that doesn't exist in v0.1.11. 

If you have a backup, I would recommend restoring from that with v0.1.11.

If you don't have a backup, then the solution is much hackier. You'll need to use the latest tag and instead change the migration to be compatible with it. The first issue you encountered is because the current nightly uses `vector_cos_ops` rather than `cosine_ops`. You can make this change in these files in the `immich-server` container: 

 - `/usr/src/app/dist/infra/migrations/1700713994428-AddCLIPEmbeddingIndex.js`
 - `/usr/src/app/dist/infra/migrations/1700714033632-AddFaceEmbeddingIndex.js`

If you're not sure how to edit these files in the container, let me know and I can try to make it into a command you can run.

Also be aware that you shouldn't try to change your CLIP model with this until we officially upgrade to v0.1.12, because you'll run into the `cosine_ops` issue again.

**Comment by IDDQD69** [comment]
Already said here, but `tensorchord/pgvecto-rs:pg14-latest` did not work for me, but `tensorchord/pgvecto-rs:pg14-v0.1.11` works.

**Comment by Teonyr** [comment]
Thanks for your help.
I'm sorry, that I messed this one up by a) not exactly using the stated version of `tensorchord/pgvecto-rs:pg14-v0.1.11` at first and b) wrongly assume, that -latest would mean latest stable not latest nightly.

I now fixed the issue by creating a completely new postgres-16 server with `tensorchord/pgvecto-rs:pg16-v0.1.11` and doing a `pg_dumpall pg15-backup` on my postgres-15 and `psql -f pg15-backup postgres` on my postgres-16.
